website/docs/core-concepts/stacks/yaml-functions/include.mdx

```diff
@@ -13,6 +13,119 @@ import File from '@site/src/components/File'
+
+Atmos resolves the `!include` functions during the initial loading of YAML files from the local filesystem or remote sources,
+injecting the contents of the referenced files directly into the current location. This applies to all stack configs.
+Included data (JSON, YAML, HCL, or text) is parsed and converted into the appropriate
+type (`string`, `boolean`, `map`, `list`, etc.) before being assigned to its place in the configuration.
+
+## Supported File Formats
+
+The `!include` function supports the following local file formats:
+  - absolute paths (e.g., `/Users/me/Documents/vars.yaml`)
+  - paths relative to the current Atmos manifest where the `!include` function is executed (e.g., `./vars.yaml`, `../config/vars.yaml`)
+  - paths relative to the [`base_path`](/cli/configuration/#base-path) defined in `atmos.yaml` CLI config file (e.g., `stacks/catalog/vpc/defaults.yaml`)
+
+To download remote files from different sources, the `!include` function uses [`go-getter`](https://github.com/hashicorp/go-getter)
+(used by [Terraform](https://www.terraform.io/) for downloading modules) and supports the following protocols:
+
+- `tar` - Tar files, potentially compressed (tar.gz, tar.bz2, etc.)
+- `zip` - Zip files
+- `http` - HTTP URLs
+- `https` - HTTPS URLs
+- `git` - Git repositories, which can be accessed via HTTPS or SSH
+- `hg` - Mercurial repositories, accessed via HTTP/S or SSH
+- `s3` - Amazon S3 bucket URLs
+- `gcs` - Google Cloud Storage URLs
+- `oci` - Open Container Initiative (OCI) images
+- `scp` - Secure Copy Protocol for SSH-based transfers
+- `sftp` - SSH File Transfer Protocol
+
+## Key Benefits of `!include`
+
+The `!include` directive enables modular, reusable configuration patterns across all Atmos YAML files — both CLI and stack configurations.
+This has several important implications:
+
+- **Modularization & Reuse**  
+  `!include` supports a [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) approach by allowing shared configuration fragments to be stored in local or remote
+   files and reused across multiple stacks or components. You can use it in place of YAML anchors, which don't work across files.
+
+- **Preprocessing for Inheritance**  
+  Includes are resolved *before* Atmos processes stacks and components, enabling powerful [inheritance](/core-concepts/stacks/inheritance/) and deep-mergin
+  behaviors using fully expanded configuration data.
+
+- **Adopting Existing Terraform/OpenTofu Root Modules as Atmos Components**  
+  If you're already managing your root modules using `.tfvars` files — for example, separate files for `dev`, `staging`,
+  and `prod` — you can reference them directly in Atmos using `!include`. This makes it easy to adopt Atmos without
+  having to translate all variables into Atmos stacks.
+
+### Example: Referencing Different TFVAR Files Per Environment
+
+For example, instead of rewriting existing `.tfvars` varfiles into inline YAML, Atmos lets you bring them in as-is.
```

Comment:
```suggestion
For example, instead of rewriting existing `.tfvars` varfiles into inline YAML, Atmos lets you bring them in as-is into your [Stacks](/core-concepts/stacks/).
```